tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -546,6 +546,34 @@ void test_VSO_226914_word_boundaries() {
     aWordAny.should_search_fail("aa", match_not_bow | match_not_eow);
 }
 
+void test_GH_993_regex_character_class_case_insensitive_search() {
+    {
+        const wstring subject = L" Copyright";
+        const test_wregex case_regex(&g_regexTester, LR"([a-z][a-z])", ECMAScript);
+        const test_wregex icase_regex(&g_regexTester, LR"([a-z][a-z])", ECMAScript | icase);
+
+        case_regex.should_search_match(subject, L"op");
+        icase_regex.should_search_match(subject, L"Co");
+    }
+
+    {
+        const wstring subject = L"blahZblah";
+        const test_wregex Z_case_regex(&g_regexTester, LR"([Z])", ECMAScript);
+        const test_wregex Z_icase_regex(&g_regexTester, LR"([Z])", ECMAScript | icase);
+        const test_wregex z_case_regex(&g_regexTester, LR"([z])", ECMAScript);
+        const test_wregex z_icase_regex(&g_regexTester, LR"([z])", ECMAScript | icase);
+
+        Z_case_regex.should_search_match(subject, L"Z");
+        Z_icase_regex.should_search_match(subject, L"Z");

```

Comment:
I'd like to see one more thing for comprehensive test coverage - verifying that `[Z]` matches `z` with `icase`.